### DISCUSSION TITLE
Taking a screenshot programatically when using the null video driver

### DISCUSSION THREAD

**Discussion by michalc** [OpenTTD/OpenTTD#12452] [Q&A]
When using the null video driver with a set number of ticks, is there some way to take a screenshot at the end / regular screenshots, say at the same time autosave happens?

The context is https://github.com/michalc/OpenTTDLab: it can run OpenTTD repeatedly and extract results from the savegame files, but I would like it to also be able to extract screenshots in order to investigate output more visually, without actually having to open up OpenTTD manually.

I'm thinking maybe a gamescript could do it somehow, but not quite sure where to begin...

**Comment by PeterN** [comment]
No, with the null video driver nothing is drawn at all.

**Reply by michalc** [reply]
Ah a shame - I was hoping it would get drawn to some virtual screen somewhere...

... so am wondering on an alternative. Can a screenshot be extracted programatically otherwise, say be running with a different video driver?

**Reply by michalc** [reply]
So made some progress in that approach in https://github.com/michalc/OpenTTDLab/pull/146/files

- It starts OpenTTD using the default video driver
- Loading the save game to take a screenshot of
- Then in `game_start.scr' it runs `screenshot giant` followed by `quit`

It does result in a black OpenTTD window opening up for a few moments for me (on macOS), and an message in the console: "dbg: [driver] Could not change to foreground application. Error -50"
